docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -231,6 +231,50 @@ var finder = new AboutToExpireSecretFinder(TimeSpan.FromDays(2), secretClient);
 
 This approach is useful when you would like to consolidate the dependency creation and share the client between multiple consuming classes.
 
+## Explore the extension methods in Azure Resource Manager SDKs
+
+For fundamental concepts of Azure Resource Manager SDKs, see [Key Concepts of Azure Resource Manager SDKs](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/resourcemanager/Azure.ResourceManager#key-concepts).
+
+In Azure Resource Manager SDKs, we design the clients (resources and collections) to have their hierarchical structures to emphasize their relationship between each other. To achieve that, extension methods are widely used because the parent resource of one resource might not exist in the same SDK library.
+
+For instance, in `Azure.ResourceManager.Compute` namespace, there is a resource `VirtualMachineResource` which stands for a virtual machine resource on Azure, and its parent is `ResourceGroupResource` in `Azure.ResourceManager` namespace, and we could get the `VirtualMachineCollection` on the `ResourceGroupResource` in this way:
```

Comment:
```suggestion
For example, an Azure virtual machine resizes in an Azure resource group.   The `Azure.ResourceManager.Compute` namespace models these types as `VirtualMachineResource` and `ResourceGroupResource`, respectively.  To query the virtual machines for a resource group, you would write:
```

Reply:
`ResourceGroupResource` class lives in `Azure.ResourceManager` assembly, I added this information in this line as well